integration-tests/src/test/java/com/datastax/oss/driver/api/core/cql/PagingIterableSpliteratorIT.java

```diff
@@ -0,0 +1,133 @@
+ */
+package com.datastax.oss.driver.api.core.cql;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import com.datastax.oss.driver.api.core.CqlSession;
+import com.datastax.oss.driver.api.core.config.DefaultDriverOption;
+import com.datastax.oss.driver.api.core.config.DriverExecutionProfile;
+import com.datastax.oss.driver.api.testinfra.ccm.CcmRule;
+import com.datastax.oss.driver.api.testinfra.session.SessionRule;
+import com.datastax.oss.driver.categories.ParallelizableTests;
+import com.datastax.oss.driver.shaded.guava.common.collect.Lists;
+import com.tngtech.java.junit.dataprovider.DataProvider;
+import com.tngtech.java.junit.dataprovider.DataProviderRunner;
+import com.tngtech.java.junit.dataprovider.UseDataProvider;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Spliterator;
+import java.util.stream.StreamSupport;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.rules.RuleChain;
+import org.junit.rules.TestRule;
+import org.junit.runner.RunWith;
+
+@RunWith(DataProviderRunner.class)
+@Category(ParallelizableTests.class)
+public class PagingIterableSpliteratorIT {
+
+  private static CcmRule ccm = CcmRule.getInstance();
+
+  private static SessionRule<CqlSession> sessionRule = SessionRule.builder(ccm).build();
+
+  @ClassRule public static TestRule chain = RuleChain.outerRule(ccm).around(sessionRule);
+
+  @BeforeClass
+  public static void setupSchema() {
+    sessionRule
+        .session()
+        .execute(
+            SimpleStatement.builder(
+                    "CREATE TABLE IF NOT EXISTS test (k0 int, k1 int, v int, PRIMARY KEY(k0, k1))")
+                .setExecutionProfile(sessionRule.slowProfile())
+                .build());
+    PreparedStatement prepared =
+        sessionRule.session().prepare("INSERT INTO test (k0, k1, v) VALUES (?, ?, ?)");
+    for (int i = 0; i < 20_000; i += 1_000) {
+      BatchStatementBuilder batch = BatchStatement.builder(DefaultBatchType.UNLOGGED);
+      for (int j = 0; j < 1_000; j++) {
+        int n = i + j;
+        batch.addStatement(prepared.bind(0, n, n));
+      }
+      sessionRule.session().execute(batch.setExecutionProfile(sessionRule.slowProfile()).build());
+    }
+  }
+
+  @Test
+  @UseDataProvider("pageSizes")
+  public void should_consume_spliterator(int pageSize, boolean parallel) throws Exception {
+    CqlSession session = sessionRule.session();
+    DriverExecutionProfile profile =
+        session
+            .getContext()
+            .getConfig()
+            .getDefaultProfile()
+            .withInt(DefaultDriverOption.REQUEST_PAGE_SIZE, pageSize);
+    ResultSet result =
+        session.execute(
+            SimpleStatement.newInstance("SELECT v FROM test where k0 = 0")
+                .setExecutionProfile(profile));
+    Spliterator<Row> spliterator = result.spliterator();
+    if (pageSize > 20_000) {
```

Comment:
why we are using `20_000` as a threshold value?

Reply:
I just picked a number that is big enough to exercise spliterators in a large data stream.